### PR TITLE
Update photom step reference file docs

### PR DESCRIPTION
**Pull request by hbushouse**
Updated the reference file information in the `photom` step docs, in partial fulfillment of #2675.

### CODE REVIEW

**Comment by hbushouse** [comment]
@dmggh Good eye, but that appears to be just an artifact of the way the page gets rendered in the github view (because of the use of the asterisk in "char*15"). It renders OK when built with sphinx.

**Comment by dmggh** [comment]
Ok, good to know (but I would have expected that effect to also show up elsewhere). Your changes looked fine otherwise.

________________________________

**Comment by hbushouse** [comment]
I've gone ahead and fixed it anyway (escape the *), just in case. It'll show up in the next commit.

File: docs/jwst/photom/reference_files.rst [hunk 1]
```diff
@@ -85,35 +228,132 @@ used in the DQ array. The PIXSIZ extension contains a 2D array of pixel
...
+reference file.
+
+NIRSpec Fixed-Slit Mode
+^^^^^^^^^^^^^^^^^^^^^^^
+
+:Data model: `~jwst.datamodels.NirspecSlitAreaModel`
+
+The BINTABLE extension has EXTNAME='AREA' and has column characteristics
+shown below. There is one row for each of the 5 fixed slits, with ``slit_id``
+values of "S200A1", "S200A2", "S400A1", "S200B1", and "S1600A1". The pixel
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply docs/jwst/photom/reference_files.rst]: Updated based on info from James, who says it is the nominal area of any pixel illuminated by the slit, computed as width of slit in dispersion direction and width of pixel in cross-dispersion direction.

File: docs/jwst/photom/reference_files.rst [hunk 2]
```diff
@@ -1,79 +1,222 @@
...
+=======  ========  =====  ===========  =========
+SCI      IMAGE       2    1032 x 1024  float
+ERR      IMAGE       2    1032 x 1024  float
+DQ       IMAGE       2    1032 x 1024  integer
+PIXSIZ   IMAGE       2    1032 x 1024  float
+DQ_DEF   BINTABLE    2    TFIELDS = 4  N/A
+=======  ========  =====  ===========  =========
 
 The SCI extension contains a 2D array of spectral sensitivity factors
 corresponding to each pixel in a 2D MRS slice image. The sensitivity factors
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply docs/jwst/photom/reference_files.rst]: Updated in latest commit to read as "(DN/sec)/(mJy/pixel)".

**Pull request merged**